Source/missiles.cpp

```diff
@@ -4911,22 +4799,21 @@ void MI_Flamec(int i)
 
 void MI_Cbolt(int i)
 {
-	int md;
 	int bpath[16] = { -1, 0, 1, -1, 0, 1, -1, -1, 0, 0, 1, 1, 0, 1, -1, 0 };
 
 	missile[i]._mirange--;
 	if (missile[i]._miAnimType != MFILE_LGHNING) {
 		if (missile[i]._miVar3 == 0) {
-			md = (missile[i]._miVar2 + bpath[missile[i]._mirnd]) & 7;
+			Direction md = (Direction)((missile[i]._miVar2 + bpath[missile[i]._mirnd]) & 7);

```

Comment:
Here is another way this could be written with out the need for a lot of casting and bit operations:

```suggestion
Direction md = static_cast<Direction>(missile[i]._miVar2);
switch (missile[i]._mirnd) {
	case 0:
	case 3:
	case 6:
	case 7:
	case 14:
		md = left[md];
		break;
	case 2:
	case 5:
	case 10:
	case 11:
	case 13:
		md = right[md];
		break;
}
```

Reply:
I don't understand what `bpath` refers to?

Maybe a subfunction with a name like `ApplyRandomDirection()` could be better?

Reply:
... hadn't seen your change, yeah that is nicer :)